### PR TITLE
RA-1134 - Add a Register Patient button to the Find Patients page

### PR DESCRIPTION
**Pull request by ssmusoke**
add override on the register patient page controller to use the custom defined app although this does not change the app in the url

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/registrationapp/RegistrationAppUiUtils.java [hunk 1]
```diff
@@ -105,4 +106,35 @@ public static void validateLatitudeAndLongitudeIfNecessary(PersonAddress address
...
+	public static String getCustomRegistrationAppId() {
+		// check if there is a custom registration app registered then use that
+		String customRegistratonAppId = Context.getAdministrationService().getGlobalProperty(RegistrationCoreConstants.GP_CUSTOM_REGISTRATION_APPID);
+		if (org.apache.commons.lang3.StringUtils.isNotBlank(customRegistratonAppId)) {
+			return customRegistratonAppId;
+		}
+		
+		return null;
+	}
+
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/registrationapp/RegistrationAppUiUtils.java]: How about merging these into one method that returns the configured appId, if none is set you return the default. If you want to keep both methods for some reason, I would rename the second one `getDefaultRegistrationAppId()`

File: omod/src/main/java/org/openmrs/module/registrationapp/page/controller/RegisterPatientPageController.java [hunk 2]
```diff
@@ -36,6 +39,15 @@ public void get(UiSessionContext sessionContext, PageModel model,
     }
 
     public void addModelAttributes(PageModel model, Patient patient, AppDescriptor app, PatientIdentifierType primaryIdentifierType, String breadcrumbOverride) throws Exception {
+        // check if the appId needs to be replaced with the custom one defined as a global property
+        if (RegistrationAppUiUtils.getRegistrationAppId() != app.getId()) {
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/registrationapp/page/controller/RegisterPatientPageController.java]: I think these are Strings, so you might want to use equals method

File: pom.xml [hunk 3]
```diff
@@ -44,7 +44,7 @@
         <emrapiVersion>1.18</emrapiVersion>
         <calculationVersion>1.2</calculationVersion>
         <providerManagementVersion>2.5.0</providerManagementVersion>
-        <registrationcoreVersion>1.6</registrationcoreVersion>
+        <registrationcoreVersion>1.7-SNAPSHOT</registrationcoreVersion>
```
Comment (wluyima) [inline pom.xml]: Do you still need this change? Now that there is no longer a change in the registration-core module?

File: omod/src/main/java/org/openmrs/module/registrationapp/page/controller/RegisterPatientPageController.java [hunk 4]
```diff
@@ -36,6 +33,15 @@ public void get(UiSessionContext sessionContext, PageModel model,
     }
 
     public void addModelAttributes(PageModel model, Patient patient, AppDescriptor app, PatientIdentifierType primaryIdentifierType, String breadcrumbOverride) throws Exception {
+        // check if the appId needs to be replaced with the custom one defined as a global property
+        if (RegistrationAppUiUtils.getDefaultRegistrationAppId().equals(app.getId())) {
+            // replace the app with the one configured for registration
+            AppDescriptor customApp = Context.getService(AppFrameworkService.class).getApp(RegistrationAppUiUtils.getDefaultRegistrationAppId());
+            if (customApp != null) {
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/registrationapp/page/controller/RegisterPatientPageController.java]: Can this ever be null? I thought if there is nothing configured, I thought the method call above returns the default.

**Comment by ssmusoke** [comment]
@wluyima I am actually thinking that we do not need this change anymore since the registration app is provided as an override to the Find Patient app

**Pull request closed without merging**